devdoc/channel.md

```diff
@@ -6,21 +6,24 @@
 
 Consumers of data call `channel_pull` to register a `PULL_CALLBACK` function to be called when data becomes available.
 
-Producers of data call `channel_push` to notify the channel that data is available and provide a `PUSH_CALLBACK` to be called when the data has been consumed.
+Producers of data call `channel_push` to notify the channel that data is available and provide a `PUSH_CALLBACK` to be called when the data has been delivered to the consumer.
 
 `channel_pull` and `channel_push` can be called from different threads in any order. The callbacks are executed on threadpool threads.
 
 ```mermaid
 stateDiagram-v2
-    B: OP_QUEUE=[]
-    N: OP_QUEUE=[push_op,...]
-    R: OP_QUEUE=[pull_op,...]
-    E: OP_QUEUE=[]
+    B: CREATED
+    O: OPEN, OP_QUEUE=[]
+    N: OPEN,OP_QUEUE=[push_op,...]
+    R: OPEN,OP_QUEUE=[pull_op,...]
+    E: OPEN,OP_QUEUE=[]
+    C: CLOSED, OP_QUEUE=[]
 
     [*] --> B: channel_create
+    B --> O: channel_open
```

Comment:
>   B --> O: channel_open [](http://example.com/codeflow?start=1&length=24)

Your geo_core change list takes advantage of the closed-> opened transition, but that's missing here.

Reply:
Added it